pootle/apps/pootle_store/unit/filters.py

```diff
@@ -6,6 +6,139 @@
 # or later license. See the LICENSE file for a copy of the license and the
 # AUTHORS file for copyright and authorship information.
 
+from django.db.models import Q
+
+from pootle_statistics.models import SubmissionTypes
+from pootle_store.models import SuggestionStates
+from pootle_store.util import FUZZY, TRANSLATED, UNTRANSLATED
+
+
+class FilterNotFound(Exception):
+    pass
+
+
+class UnitFilter(object):
+
+    def __init__(self, qs, *args, **kwargs):
+        self.qs = qs
+
+    def filter(self, unit_filter):
+        try:
+            return getattr(
+                self, "filter_%s" % unit_filter.replace("-", "_"))()
+        except AttributeError:
+            raise FilterNotFound()
+
+
+class UnitChecksFilter(UnitFilter):
+
+    def __init__(self, qs, *args, **kwargs):
+        self.qs = qs
+        self.checks = kwargs.get("checks")
+        self.category = kwargs.get("category")
+
+    def filter_checks(self):
+        if self.checks:
+            return self.qs.filter(
+                qualitycheck__false_positive=False,
+                qualitycheck__name__in=self.checks).distinct()
+        elif self.category:
+            return self.qs.filter(
+                qualitycheck__false_positive=False,
+                qualitycheck__category=self.category).distinct()
+        return self.qs.none()
+
+
+class UnitStateFilter(UnitFilter):
+    """Filter a Unit qs based on unit state"""
+
+    def filter_translated(self):
+        return self.qs.filter(state=TRANSLATED)
+
+    def filter_untranslated(self):
+        return self.qs.filter(state=UNTRANSLATED)
+
+    def filter_fuzzy(self):
+        return self.qs.filter(state=FUZZY)
+
+    def filter_incomplete(self):
+        return self.qs.filter(
+            Q(state=UNTRANSLATED) | Q(state=FUZZY))
+
+
+class UnitContributionFilter(UnitFilter):
+    """Filter a Unit qs based on user contributions"""
+
+    def __init__(self, qs, *args, **kwargs):
+        self.qs = qs
+        self.user = kwargs.get("user")
+
+    def filter_suggestions(self):
+        return self.qs.filter(
+            suggestion__state=SuggestionStates.PENDING).distinct()
+
+    def filter_my_suggestions(self):
+        if not self.user:
+            return self.qs.none()
+        return self.qs.filter(
+            suggestion__state=SuggestionStates.PENDING,
+            suggestion__user=self.user).distinct()
```

Comment:
that would work - but doesnt matter so much now as i followed your other suggestion